pilot/pkg/model/authentication.go

```diff
@@ -312,7 +313,24 @@ func SdsCertificateConfigFromResourceName(resource string) (SdsCertificateConfig
 	} else if strings.HasPrefix(resource, "file-root:") {
 		filesString := strings.TrimPrefix(resource, "file-root:")
 		split := strings.Split(filesString, ResourceSeparator)
-		if len(split) != 1 {
+		if filesString == "system" && len(split) != 1 {
+			var certFiles = []string{
+				"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+				"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
+				"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
+				"/etc/pki/tls/cacert.pem",                           // OpenELEC
+				"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+				"/etc/ssl/cert.pem",                                 // Alpine Linux
+				"/usr/local/etc/ssl/cert.pem",                       // FreeBSD
+			}
+
+			for _, cert := range certFiles {
+				if _, err := os.Stat(cert); err == nil {
```

Comment:
small perf nit: should we wrap this in a `once` like go does? Its fairly slow to check the filesystem on every call here

Reply:
+1. And the value seems not neeed to passed around, and parameteize and special casing it. It's fair to assume the proxy image's filesystem stay the same during the pod running time and keep use that value.